routes/api/questions.js

```diff
@@ -6,14 +6,22 @@ const router = express.Router();
 const models = require('../../models');
 
 router.get('/', async (req, res) => {
-  const records = await models.Question.findAll({ order: [['prompt', 'ASC']] });
+  const options = {
+    order: [['prompt', 'ASC']],
+  };
+  if (req.query.questionnaireId) {
```

Comment:
Here, we add a a "where" clause to the database query to search for Question records that belong to a given QuestionnaireId, if any...